pkg/controller/signup.go

```diff
@@ -109,7 +100,7 @@ func (s *Signup) GetHandler(ctx *gin.Context) {
 	signupResource, err := s.app.SignupService().GetSignup(userID)
 	if err != nil {
 		log.Error(ctx, err, "error getting UserSignup resource")
-		errors.AbortWithError(ctx, http.StatusInternalServerError, err, "error getting UserSignup resource")
+		errors.AbortWithStatusError(ctx, err, "error getting UserSignup resource")
```

Comment:
I guess we always want to return 500 if something goes wrong here.

Reply:
Yeah, right now only 500 will be returned. I just thought in case in the future we add some other error message this is already setup to handle it. WDYT?